app/javascript/packs/application.js

```diff
@@ -19,30 +19,34 @@ global.mapsLoadedCallback = function() {
     let instance = application.getControllerForElementAndIdentifier(map, "map");
     instance.initMap() ;
   }
-}
+};
 
 Accordion.prototype.originalSetExpanded = Accordion.prototype.setExpanded;
 Accordion.prototype.setExpanded = function (expanded, $section) {
+  // the name of the accordion as it will be categorised in GA
   const eventName = this.$module.attributes['data-track-event-name'];
 
-  if (eventName) {
-    const segment = $section.querySelector('button').id;
-
-    if (expanded && !$section.classList.contains(this.sectionExpandedClass)) {
-      sendGAEvent(eventName.value, segment, 'open');
-    }
-    else if (!expanded && $section.classList.contains(this.sectionExpandedClass)) {
-      sendGAEvent(eventName.value, segment, 'closed');
-    }
+  // the button inserted by the accordion plugin that replaces the
+  // provided span and takes its descriptive ID
+  const sectionButton = $section.querySelector('button');
+
+  if (eventName && sectionButton) {
+    const accordionExpanded = new CustomEvent('accordionToggle', {
+      bubbles: true,
+      detail: {
+        name: eventName.value,
+        section: sectionButton.id,
+        expanded: expanded
+      }
+    });
+
+    $section.dispatchEvent(accordionExpanded);
   }
 
   this.originalSetExpanded(expanded, $section);
 };
 
-window.descriptionTracker = {
-  element: descriptionSummary,
-  event: descriptionToggleEvent
-};
+window.logAccordionToggle = logAccordionToggle;
```

Comment:
could this be instead

```javascript
document.addEventListener(accordionToggle, logAccordionToggle)
```

Log accordion toggles prototype would need to reflect the expected parameters for a event handled

Another option is to export a new event handler from `log_accordion_toggle.js` separately and use that as the intersection between the existing logging function and the event listener? Ping me if thats not making much sense.